cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -137,10 +137,10 @@ class parquet_field_bool : public parquet_field {
 struct parquet_field_bool_list : public parquet_field_list<bool, FieldType::BOOLEAN_TRUE> {
   parquet_field_bool_list(int f, std::vector<bool>& v) : parquet_field_list(f, v)
   {
-    auto const read_value = [this](uint32_t i, CompactProtocolReader* cpr) {
+    auto const read_value = [&val = v](uint32_t i, CompactProtocolReader* cpr) mutable {
```

Comment:
Do you need the mutable keyword? You're capturing by reference so I would have assumed not.

Reply:
Exactly, I wanted to ask the same question otherwise LGTM

Reply:
Same comment as Yunsong here

Reply:
The `mutable` specifier is defined here: https://en.cppreference.com/w/cpp/language/lambda 

> mutable	Allows body to modify the objects captured by copy, and to call their non-const member functions.

I added the specifier since all of the lambdas call non-const member functions.
But I tried compiling it without the specifier and the 12.3 compiler seemed ok so I can technically remove it.

Reply:
Since you're capturing `val` by reference, not by copy, I don't think it's necessary. Not a big deal though.